src/main/java/uk/co/ramp/Outbreak.java

```diff
@@ -0,0 +1,339 @@
+    private final Map<Integer, CmptRecord> records = new HashMap<>();
+
+    public void setPopulation(Map<Integer, Case> population) {
+        this.population = population;
+    }
+
+    public void setContactRecords(Map<Integer, List<ContactRecord>> contactRecords) {
+        this.contactRecords = contactRecords;
+    }
+
+
+    @Autowired
+    public void setDiseaseProperties(DiseaseProperties diseaseProperties) {
+        this.diseaseProperties = diseaseProperties;
+    }
+
+    @Autowired
+    public void setStandardProperties(StandardProperties standardProperties) {
+        this.properties = standardProperties;
+    }
+
+    @Autowired
+    public void setRandomDataGenerator(RandomDataGenerator randomDataGenerator) {
+        this.rng = randomDataGenerator;
+    }
+
+
+    public Map<Integer, CmptRecord> propagate() {
+
+        generateInitialInfection();
+        runToCompletion();
+
+        return records;
+    }
+
+    private void generateInitialInfection() {
+        Set<Integer> infectedIds = infectPopulation();
+        for (Integer id : infectedIds) {
+
+            EvaluateCase evaluateCase = new EvaluateCase(population.get(id), diseaseProperties, rng);
+            evaluateCase.updateVirusStatus(EXPOSED, 0);
+
+        }
+    }
+
+    int tab = 0;
+
+    private void runToCompletion() {
+        int timeLimit = properties.timeLimit();
+        int maxContact = contactRecords.keySet().stream().max(Comparator.naturalOrder()).orElseThrow(RuntimeException::new);
+        int runTime;
+        boolean steadyState = properties.steadyState();
+        double randomInfectionRate = diseaseProperties.randomInfectionRate();
+
+        if (timeLimit <= maxContact) {
+            LOGGER.info("Not all contact data will be used");
+            runTime = timeLimit;
+            steadyState = false;
+        } else {
+            LOGGER.info("Potential for steady state soln");
+            runTime = maxContact;
+        }
+
+        boolean complete = runContactData(runTime, population, contactRecords, randomInfectionRate);
+
+        if (steadyState && !complete) {
+            runToSteadyState(runTime, timeLimit, population);
+        }
+
+        outputInfectionData(population);
+
+
+    }
+
+    private void outputInfectionData(Map<Integer, Case> population) {
+        List<Case> infections = new ArrayList<>();
+        for (Case c : population.values()) {
+            if (c.status() != SUSCEPTIBLE) {
+                infections.add(c);
+            }
+        }
+
+        infections.sort(Comparator.comparingInt(Case::exposedTime));
+
+        Map<Integer, Set<Integer>> infectors = new HashMap<>();
+
+        for (Case c : infections) {
+            infectors.putIfAbsent(c.exposedBy(), new HashSet<>());
+
+            Set<Integer> var = infectors.get(c.exposedBy());
+
+            var.add(c.id());
+            infectors.put(c.exposedBy(), var);
+        }
+
+        // initial infectees
+        Set<Integer> infectees = infectors.get(-9);
```

Comment:
-9 is this a magic number? If it is can we declare it as a static and move it to top as a class variable?